components/client/src/main/java/com/hotels/styx/client/loadbalancing/strategies/PowerOfTwoStrategy.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (C) 2013-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.client.loadbalancing.strategies;
+
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.hotels.styx.api.Environment;
+import com.hotels.styx.api.client.ActiveOrigins;
+import com.hotels.styx.api.client.ConnectionPool;
+import com.hotels.styx.api.client.RemoteHost;
+import com.hotels.styx.api.client.loadbalancing.spi.LoadBalancingStrategy;
+import com.hotels.styx.api.client.loadbalancing.spi.LoadBalancingStrategyFactory;
+import com.hotels.styx.api.configuration.Configuration;
+
+import java.util.List;
+import java.util.Random;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.StreamSupport.stream;
+
+
+/**
+ * A Power Of Two load balancing strategy.
+ *
+ * Randomly choose two origins, and return the one with less ongoing requests.
+ */
+public class PowerOfTwoStrategy implements LoadBalancingStrategy {
+    private ActiveOrigins activeOrigins;
+    private Random rng;
+
+    public PowerOfTwoStrategy(ActiveOrigins activeOrigins) {
+        this(activeOrigins, new Random());
+    }
+
+    @VisibleForTesting
+    PowerOfTwoStrategy(ActiveOrigins activeOrigins, Random rng) {
+        this.activeOrigins = activeOrigins;
```

Comment:
These parameters should be null-checked. Please add these checks *before* submitting a pull request.